contracts/utils/VaultQueue.sol

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.4;
+
+import "hardhat/console.sol";
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ReentrancyGuardUpgradeable
+} from "@openzeppelin/contracts-upgradeable/security/ReentrancyGuardUpgradeable.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+    SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ISAVAX} from "../interfaces/ISAVAX.sol";
+import {IRibbonVault, IDepositContract} from "../interfaces/IRibbon.sol";
+import {Vault} from "../libraries/Vault.sol";
+import {DSMath} from "../vendor/DSMath.sol";
+
+interface IVaultQueue {
+    enum TransferAction {INTERVAULT, WITHDRAW, SKIP_ROUND}
+    struct Transfer {
+        address creditor;
+        address srcVault;
+        address dstVault;
+        address depositContract;
+        TransferAction action;
+        uint256 amount;
+        uint256 timestamp;
+    }
+}
+
+contract VaultQueue is
+    Initializable,
+    OwnableUpgradeable,
+    ReentrancyGuardUpgradeable
+{
+    using SafeERC20 for IERC20;
+
+    mapping(address => IVaultQueue.Transfer[]) public qTransfer;
+    mapping(address => uint256) public totalAmount;
+    address[] public vaults;
+
+    address public wethVault = 0x25751853Eab4D0eB3652B5eB6ecB102A2789644B;
+    address public stEthVault = 0x53773E034d9784153471813dacAFF53dBBB78E8c;
```

Comment:
Probably should pass these 2 vars in from a constructor instead of hardcoding them

Reply:
I was considering this but the vault behavior on other chains is different.  e.g. AVAX and sAVAX.

Reply:
done